python/lsst/meas/base/plugins.py

```diff
@@ -68,6 +71,103 @@
                     TransformClass=bl.ApertureFluxTransform, executionOrder=BasePlugin.FLUX_ORDER)
 
 # --- Single-Frame Measurement Plugins ---
+class SingleFrameVarianceConfig(SingleFramePluginConfig):
+        scale = lsst.pex.config.Field(dtype=float, default=5.0, optional=True,
+                                      doc="scale factor to apply to shape for aperture")
+        mask = lsst.pex.config.ListField(doc="mask planes to ignore", dtype=str,
+                                         default=["DETECTED", "DETECTED_NEGATIVE"])
+
+@register("base_Variance")
+class SingleFrameVariancePlugin(SingleFramePlugin):
+    '''
+    Calculate the median variance within a Footprint scaled from the object shape so
+    the value is not terribly influenced by the object and instead represents the
+    variance in the background near the object.
+    '''
+
+    ConfigClass = SingleFrameVarianceConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.FLUX_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+        self.varValue = schema.addField(name + '_value', type="D", doc="variance at object position")
+        self.varFlag = schema.addField(name + '_flag', type="Flag", doc="Set to 1 for any fatal Failure")
+
+    def measure(self, measRecord, exposure):
+        # Promote the bounding box of the footprint to type D to ensure
+        # the ability to compare (they may be of mixed types I and D)
+        center = measRecord.getCentroid()
+        fpBbox = lsst.afw.geom.Box2D(measRecord.getFootprint().getBBox())
+        # check to insure that the center exists and that it is contained within the footprint
+        # catch bad centroiding
+        if not center or not fpBbox.contains(center):
+            raise Exception("The measRecord has no center, or the center is outside the footprint")
```

Comment:
measRecord is just a variable name -- it's an artefact of your implementation, and likely means nothing to the end user. Maybe better to refer to it just as a "measurement" or a "source record" or something?

Also: you can trivially tell if the problem is that the measurement has no centre or if the centre is outside the Footprint. The end user likely can't. Would it be worth being explicit about what has caused the failure?

Also also: capital-F-ootprint.